converters/kernel.go

```diff
@@ -7,54 +7,72 @@ import (
 	"strconv"
 )
 
-// Convert - method for converting from string into any other types
-func Convert(source reflect.Value, destination reflect.Value) (reflect.Value, error) {
+/*ConvertBetweenPrimitiveTypes - method for converting from any of base types into any of base types,
+// like string, bool, int, int8, int16. int32, int64
+*/
+func ConvertBetweenPrimitiveTypes(source reflect.Value, destination reflect.Value) (reflect.Value, error) {
 	fmt.Println("Level: Debug. Message: start converting source: ", source.Kind().String(), " destination: ", destination.Kind().String())
 	switch destination.Kind() {
-	case reflect.Int16, reflect.Int32, reflect.Int64:
+	case reflect.Int:
 		return convertToInt(source, destination)
+	case reflect.Int8:
+		return convertToInt8(source, destination)
+	case reflect.Int16:
+		return convertToInt16(source, destination)
+	case reflect.Int32:
+		return convertToInt32(source, destination)
+	case reflect.Int64:
+		return convertToInt64(source, destination)
 	case reflect.String:
-		return source, nil
+		return convertToString(source, destination)
 	case reflect.Bool:
 		return convertToBool(source, destination)
+	default:
+		return reflect.Zero(nil), errors.New("can not converted to this type: " + destination.Kind().String())
 	}
-	return reflect.Zero(nil), errors.New("not implementeded")
 }
 
-// depreceted
 func convertToInt(source reflect.Value, destination reflect.Value) (reflect.Value, error) {
 	switch source.Kind() {
 	case reflect.String:
-		convertByReflection := source.Convert(destination.Type())
-		fmt.Println("Level: Debug. Message: Convert by reflection: ", convertByReflection.Kind(), " can set?:", convertByReflection.CanSet())
+		convertByReflection := source.String()
+		fmt.Println("Level: Debug. Message: start convert value ", convertByReflection, " into int type")
+
+		// fmt.Println("Level: Debug. Message: Convert by reflection: ", convertByReflection.Kind(), " can set?:", convertByReflection.CanSet())
 		convertedValue, errorConvert := strconv.ParseInt(source.String(), 10, 64)
```

Comment:
Use `convertByReflection` variable instead of repeatedly getting it from source

Reply:
in this issue i will not fixed it because currently all of job for this package in issue 13